### PR TITLE
增加了HTTP/2模式

### PR DESCRIPTION
**Pull request by t1935111**
增加http2
支持tls 1.3

### CODE REVIEW

**Comment by Mygod** [comment]
I don't see the benefit of this, as the added latency from TLS handshake easily dominates the overall efficiency.

File: main.go [hunk 1]
```diff
@@ -132,11 +134,30 @@ func generateConfig() (*core.Config, error) {
 			},
 		}
 		connectionReuse = true
+		if !*tlsEnabled {
+			logInfo("mode: websocket")
+		} else {
+			logInfo("mode: websocket tls")
+		}
+		break
```
Comment (Mygod) [inline main.go]: Why is `break` needed?

File: main.go [hunk 2]
```diff
@@ -132,11 +134,30 @@ func generateConfig() (*core.Config, error) {
...
+		logInfo("mode: quic tls")
+		break
+	case "http":
+		transportSettings = &http.Config{
+			Path: *path,
+			Host: []string {
+				*host,
+			},
+		}
+		*tlsEnabled = true
```
Comment (Mygod) [inline main.go]: Why do you enforce tls?

**Comment by t1935111** [comment]
> I don't see the benefit of this, as the added latency from TLS handshake easily dominates the overall efficiency.

优点：增加http2模式，仅仅是为了增加服务的隐蔽性。我本身有在做网络管理的工作，对于一个正常网站，流量特征不应该充斥着大量的ws流量（ws一般更多用在长连接、单次传递少量数据）。我也考虑过使用quic，但是我发现网络商对于quic的支持，暂时还不是很友好。

延迟：握手延迟，我测试使用差不多3个月，和ws比较，暂时还感知不到之间的差异。

File: main.go (hunk 2 above)
Comment (t1935111) [reply main.go]: 虽然http2没有规定要基于tls部署，不过从现在的网站部署趋势来看，http2部署在tls下的占绝大多数。因为我是这样考虑，所以我就强制开启tls选项。

File: main.go (hunk 1 above)
Comment (t1935111) [reply main.go]: 我承认因为是我学艺未深，从别的语言转过来的诟病。抱歉抱歉，真的不需要break。

File: main.go [hunk 3]
```diff
@@ -39,8 +40,9 @@ import (
 	"v2ray.com/core/common/platform/filesystem"
 )
 
-var (
-	VERSION    = "custom"
+const (
+	ver       = "1.1.0"
+	buildData = "20190226"
```
Comment (Mygod) [inline main.go]: Do not change this.

**Comment by MaxMadcc** [comment]
> > I don't see the benefit of this, as the added latency from TLS handshake easily dominates the overall efficiency.
> 
> 优点：增加http2模式，仅仅是为了增加服务的隐蔽性。我本身有在做网络管理的工作，对于一个正常网站，流量特征不应该充斥着大量的ws流量（ws一般更多用在长连接、单次传递少量数据）。我也考虑过使用quic，但是我发现网络商对于quic的支持，暂时还不是很友好。
> 
> 延迟：握手延迟，我测试使用差不多3个月，和ws比较，暂时还感知不到之间的差异。

请问能编译通过吗

File: main.go (hunk 3 above)
Comment (t1935111) [reply main.go]: 已经修改回来

Comment (t1935111) [reply main.go]: > Do not change this.

已经按照要求修改回来了

**Comment by madeye** [comment]
@Mygod Any feedback?

**Comment by Mygod** [comment]
As mentioned in the previous review, I don't think there is any reason to use HTTP/2 over websocket. TLS 1.3 will be added separately. (also can anyone explain why new releases are authored by v2ray contributors instead?)

**Comment by klzgrad** [comment]
> HTTP/2 mode is simply a PUT request with no fancy connection multiplexing

This claim is mostly false. H/2 itself implements connection multiplexing. Unless V2ray is using H/2 in a really bad way.

**Pull request closed without merging**